bulldozer/merge.go

```diff
@@ -187,51 +200,53 @@ func calculateCommitMessage(ctx context.Context, pullCtx pull.Context, client *g
 			}
 		}
 	case SummarizeCommits:
-		summarizedMessages, err := summarizeCommitMessages(ctx, pullCtx, client)
+		summarizedMessages, err := summarizeCommitMessages(ctx, pullCtx)
 		if err != nil {
-			return "", errors.Wrap(err, "failed to collect pull request commit messages")
+			return "", errors.Wrap(err, "failed to summarize pull request commit messages")
 		}
-
 		commitMessage = summarizedMessages
 	case EmptyBody:
-	default:
 	}
 
 	return commitMessage, nil
 }
 
-func summarizeCommitMessages(ctx context.Context, pullCtx pull.Context, client *github.Client) (string, error) {
-	var builder strings.Builder
-	repositoryCommits, err := allCommits(ctx, pullCtx, client)
-	if err != nil {
-		return "", errors.Wrapf(err, "cannot list commits for %q", pullCtx.Locator())
+func calculateCommitTitle(ctx context.Context, pullCtx pull.Context, option SquashOptions) (string, error) {
+	var title string
+	switch option.Title {
+	case PullRequestTitle:
+		prTitle, err := pullCtx.Title(ctx)
+		if err != nil {
+			return "", err
+		}
+		title = prTitle
+	case FirstCommitTitle:
+		commits, err := pullCtx.Commits(ctx)
+		if err != nil {
+			return "", err
+		}
+		// TODO(bkeyes): check this assumption
+		// commits are ordered from newest to oldest
+		firstCommit := commits[len(commits)-1]
+		title = strings.SplitN(firstCommit.Message, "\n", 1)[0]
+	case GithubDefaultTitle:
```

Comment:
if default, it'll return an empty string? sometimes i do wish there was an optional that wasnt null

Reply:
Yeah, it will return the default value of `title`, which is empty, which GitHub will then interpret as the default/unset behavior.